### PR TITLE
[Regression Tests][hipCUB] Implementation of regression subtest for emulation

### PR DESCRIPTION
**Pull request by NguyenNhuDi**
The implementation of regression test for emulation in rocThrust.

This is a continuation of (#419 ), do not merge until (#419  ) is merged and closed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,13 +5,21 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
 ## (Unreleased) hipCUB-x.x.x for ROCm 6.4.0
 
 ### Added
+* Added regression tests to `rtest.py`. Regression tests are a subset of tests that caused hardware problems for past emulation environments.
+  * Can be run with `python rtest.py [--emulation|-e|--test|-t]=regression`
```
Comment (spolifroni-amd) [inline CHANGELOG.md]: If you combine with the point above, you can remove this line.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,13 +5,21 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
 ## (Unreleased) hipCUB-x.x.x for ROCm 6.4.0
 
 ### Added
+* Added regression tests to `rtest.py`. Regression tests are a subset of tests that caused hardware problems for past emulation environments.
```
Comment (spolifroni-amd) [inline CHANGELOG.md]:
```suggestion
* Added regression tests to `rtest.py`. These tests recreate scenarios that have caused hardware problems in past emulation environments. Use `python rtest.py [--emulation|-e|--test|-t]=regression` to run these tests.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,13 +5,21 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
...
+  * Can be run with `python rtest.py [--emulation|-e|--test|-t]=regression`
 * Added `ForEach`, `ForEachN`, `ForEachCopy`, `ForEachCopyN` and `Bulk` functions to have parity with CUB.
 * Added the `hipcub::CubVector` type for CUB parity.
+* Added `--emulation` option for `rtest.py`
```
Comment (spolifroni-amd) [inline CHANGELOG.md:11]: This is redundant since it was covered in the first point. It can be removed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,13 +5,21 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
...
+* Unit tests can be ran with `[--emulation|-e|--test|-t]=<test_name>`
```
Comment (spolifroni-amd) [inline CHANGELOG.md]: Which part of this was added? The --emulation option or something else? Or is it the script itself? In which case we can rewrite the first point about regression tests.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,13 +5,21 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
...
+
+### Changed
+* Edited smoke test options, which runs a subset of the unit tests and ensuring that less than 2gb of VRAM will be used
```
Comment (spolifroni-amd) [inline CHANGELOG.md]: Is it that the smoke tests have been changed to use less than 2GB of VRAM? Or is it that the smoke tests are now run in different way? That needs to be clarifued.

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,13 +5,21 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
...
+  * Smoke tests can be ran with `[--emulation|-e|--test|-t]=smoke`
+* `--test|-t` is no longer a required flag for `rtest.py`. Instead, the user can use either `--emulation|-e` or `--test|-t`, but not both.
```
Comment (spolifroni-amd) [inline CHANGELOG.md]: I would put the change to rtest.py first.

```suggestion
### Changed
* The `rtest.py` options have changed.  `rtest.py` is now run with at least either `--test|-t` or `--emulation|-e`, but not both options.
```

**Review by spolifroni-amd — CHANGES_REQUESTED** [review]
The changes to changelog need to be fixed. Might as well get ahead of these now so there's less work to do before release.

File: CHANGELOG.md (hunk 1 above)
Comment (NguyenNhuDi) [reply CHANGELOG.md:9]: To keep things consistent with other changelogs for other primitive library, I think we should keep it as two lines.

File: CHANGELOG.md (hunk 4 above)
Comment (NguyenNhuDi) [reply CHANGELOG.md]: The --emulation option

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,16 +6,14 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
 
 ### Added
 * Added regression tests to `rtest.py`. These tests recreate scenarios that have caused hardware problems in past emulation environments. Use `python rtest.py [--emulation|-e|--test|-t]=regression` to run these tests.
-  * Can be run with `python rtest.py [--emulation|-e|--test|-t]=regression`
 * Added `ForEach`, `ForEachN`, `ForEachCopy`, `ForEachCopyN` and `Bulk` functions to have parity with CUB.
 * Added the `hipcub::CubVector` type for CUB parity.
 * Added `--emulation` option for `rtest.py`
-* Unit tests can be ran with `[--emulation|-e|--test|-t]=<test_name>`
+  * Unit tests can be ran with `[--emulation|-e|--test|-t]=<test_name>`
```
Comment (spolifroni-amd) [inline CHANGELOG.md]:
```suggestion
  * Unit tests can be run with `[--emulation|-e|--test|-t]=<test_name>`
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,16 +6,14 @@ Full documentation for hipCUB is available at [https://rocm.docs.amd.com/project
...
 
 ### Changed
-* Edited smoke test options, which runs a subset of the unit tests and ensuring that less than 2gb of VRAM will be used
-  * Smoke tests can be ran with `[--emulation|-e|--test|-t]=smoke`
-* `--test|-t` is no longer a required flag for `rtest.py`. Instead, the user can use either `--emulation|-e` or `--test|-t`, but not both.
+* Changed subset of tests to be ran for smoke tests such that it will complete in faster run-time and to never exceed 2GB of vram usage. Use `python rtest.py [--emulation|-e|--test|-t]=smoke` to run these tests.
```
Comment (spolifroni-amd) [inline CHANGELOG.md]:
```suggestion
* Changed the subset of tests that are run for smoke tests such that the smoke test will complete with faster run-time and to never exceed 2GB of vram usage. Use `python rtest.py [--emulation|-e|--test|-t]=smoke` to run these tests.
```

**Review by spolifroni-amd — APPROVED** [review]
There are two small gramma changes that need to be made, but otherwise it's fine. I'm approving so you don't have to wait for me once you make the changes.

**Review by umfranzw — APPROVED** [review]
Looks good to me. The CI test failures don't appear to be caused by this change.

**Pull request merged**